features/fixtures/mazerunner/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXExternalStackElementScenario.java

```diff
@@ -0,0 +1,31 @@
+package com.bugsnag.android.mazerunner.scenarios;
+
+import android.content.Context;
+
+import com.bugsnag.android.Configuration;
+
+import android.support.annotation.NonNull;
+
+public class CXXExternalStackElementScenario extends Scenario {
+    static {
+        System.loadLibrary("bugsnag-ndk");
+        System.loadLibrary("entrypoint");
+    }
+
+    public native void crash(int counter);
+
+    public CXXExternalStackElementScenario(@NonNull Configuration config, @NonNull Context context) {
+        super(config, context);
+        config.setAutoCaptureSessions(false);
+    }
+
+    @Override
+    public void run() {
+        super.run();
+        String metadata = getEventMetaData();
+        if (metadata != null && metadata.equals("non-crashy")) {
+            return;
+        }
```

Comment:
~30 files in the `scenario` package contain the same 4 lines as above, and nearly every scenario contains `config.setAutoCaptureSessions(false);`.

I think it'd be good to think about whether there are any abstractions/sensible defaults we could use to reduce this duplication. This could be addressed as a separate ticket and shouldn't hold up the work in this PR.

Reply:
Its still helpful to know for every test case exactly what is going on, and then not fight it if for some reason its needed in a more complex case down the line.

Reply:
To add some additional context, say we refactor this into a method, what then would it save? Maybe a line at most?

```java
if (shouldNotCrash()) {
  return;
}
```

Reply:
Personally I think a method like `shouldNotCrash` would improve readability. The reader doesn't have to know what event metadata is, why it needs a null-check, or where 'non-crashy' has been set - they only need to know that the author didn't want the scenario to crash at this point.

Additionally, we wouldn't have to worry about updating 30 separate source files if we need to make a change to this feature, and could also add some brief JavaDoc to explain the convention of setting 'non-crashy' as an Environment Variable within a feature.

This doesn't need to be addressed as part of this PR but I do feel it's worth creating a ticket to address this at some future point.

Reply:
I have taken a look at this and I think that my main problem with it is the name of the variable its stored in. It's too easily confused with event metadata. If we made this called something obviously environment variable like then this code seems fine to me.